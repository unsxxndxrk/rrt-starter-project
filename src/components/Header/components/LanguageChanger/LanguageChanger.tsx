import { Languages } from "types/common/Languages";
import { enumToStringArray } from "utils/enumToStringArray";
import { useLanguageChangerStyles } from "./LanguageChanger.styles";
import { useTranslation } from "react-i18next";
import cx from "classnames";

function LanguageChanger() {
  const appLanguages = enumToStringArray(Languages);
  const classes = useLanguageChangerStyles();

  const { i18n } = useTranslation();
  const currentLanguage = i18n.language;

  const handgeLanguageChange = (nextLang: string) => {
    i18n.changeLanguage(nextLang);
  };

  return (
    <div className={classes.root}>
      {appLanguages.length &&
        appLanguages.map((l: string, key: number) => (
          <div className={classes.langWrap} key={key}>
            <div
              className={cx(classes.lang, {
                active: currentLanguage === l,
              })}
              onClick={() => handgeLanguageChange(l)}
            >
              {l}
            </div>
            {appLanguages.length - 1 !== appLanguages.indexOf(l) && <>|</>}
          </div>
        ))}
    </div>
  );
}

export { LanguageChanger };
