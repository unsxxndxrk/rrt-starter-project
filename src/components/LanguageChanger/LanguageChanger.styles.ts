import { makeStyles, Theme } from "@material-ui/core/styles";

const useLanguageChangerStyles = makeStyles((theme: Theme) => ({
  root: {
    position: "absolute",
    right: 0,
    top: "50%",
    transform: "translate(0, -50%)",
    display: "flex",
    alignItems: "center",
  },
  langWrap: {
    display: "flex",
    alignItems: "center",
  },
  lang: {
    textTransform: "uppercase",
    padding: "0 5px",
    cursor: "pointer",
    "&:hover": {
      color: theme.palette.primary.main,
    },
    "&.active": {
      color: theme.palette.primary.main,
      fontWeight: "bold",
    },
  },
}));

export { useLanguageChangerStyles };
