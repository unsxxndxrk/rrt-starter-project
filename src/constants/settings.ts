export const APP_DEFAULT_LANGUAGE: string = "en";
export const API_ENDPOINT: string =
  process.env.REACT_APP_API_ENDPOINT || window.location.origin;
