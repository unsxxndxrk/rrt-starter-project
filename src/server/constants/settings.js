const ORIGIN = "http://localhost";
const PORT = "8000";
const USERS_PATH = "./data/users.json";
const TODOS_PATH = "./data/todos.json";

exports.ORIGIN = ORIGIN;
exports.PORT = PORT;
exports.USERS_PATH = USERS_PATH;
exports.TODOS_PATH = TODOS_PATH;
