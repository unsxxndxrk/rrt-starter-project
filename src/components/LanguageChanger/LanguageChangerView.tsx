import { useLanguageChangerStyles } from "./LanguageChanger.styles";
import cx from "classnames";

type LanguageChangerViewProps = {
  appLanguages: string[];
  currentLanguage: string;
  onLanguageChange: (nextLang: string) => void;
};

function LanguageChangerView(props: LanguageChangerViewProps) {
  const { appLanguages, currentLanguage, onLanguageChange } = props;
  const classes = useLanguageChangerStyles();

  return (
    <div className={classes.root}>
      {appLanguages.length &&
        appLanguages.map((l: string, key: number) => (
          <div className={classes.langWrap} key={key}>
            <div
              className={cx(classes.lang, {
                active: currentLanguage === l,
              })}
              onClick={() => onLanguageChange(l)}
            >
              {l}
            </div>
            {appLanguages.length - 1 !== appLanguages.indexOf(l) && <>|</>}
          </div>
        ))}
    </div>
  );
}

export default LanguageChangerView;
