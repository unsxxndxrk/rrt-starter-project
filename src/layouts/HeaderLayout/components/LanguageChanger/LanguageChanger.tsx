import React from "react";
import { Languages } from "../../../../types/common/Languages";
import { enumToStringArray } from "../../../../utils/enumToStringArray";
import { useLanguageChangerStyles } from "./LanguageChanger.styles";
import { useTranslation } from "react-i18next";
import cx from "classnames";
import { uiLanguage } from "../../../../utils/storageHelpers";

const LanguageChanger = () => {
  const appLanguages = enumToStringArray(Languages);
  const classes = useLanguageChangerStyles();

  const { i18n } = useTranslation();
  const currentLanguage = uiLanguage.get();

  const handgeLanguageChange = (nextLang: string) => {
    i18n.changeLanguage(nextLang);
  };

  return (
    <div className={classes.root}>
      {appLanguages.length &&
        appLanguages.map((l: string, key: number) => (
          <div className={classes.langWrap} key={key}>
            <div
              className={cx(classes.lang, {
                active: currentLanguage === l,
              })}
              onClick={() => handgeLanguageChange(l)}
            >
              {l}
            </div>
            {appLanguages.length - 1 !== appLanguages.indexOf(l) && <>|</>}
          </div>
        ))}
    </div>
  );
};

export default LanguageChanger;
