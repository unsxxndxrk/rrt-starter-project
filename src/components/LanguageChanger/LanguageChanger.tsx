import { Languages } from "types/common/Languages";
import { enumToStringArray } from "utils/enumToStringArray";
import { useTranslation } from "react-i18next";
import { LanguageChangerView } from "./LanguageChangerView";

function LanguageChanger() {
  const { i18n } = useTranslation();

  const appLanguages = enumToStringArray(Languages);
  const currentLanguage = i18n.language;

  const handgeLanguageChange = (nextLang: string) => {
    i18n.changeLanguage(nextLang);
  };

  return (
    <LanguageChangerView
      appLanguages={appLanguages}
      currentLanguage={currentLanguage}
      onLanguageChange={handgeLanguageChange}
    />
  );
}

export { LanguageChanger };
