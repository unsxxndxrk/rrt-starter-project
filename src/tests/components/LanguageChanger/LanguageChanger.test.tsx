import { render } from "tests/settings/render";
import LanguageChanger from "components/LanguageChanger/LanguageChanger";

describe("<LanguageChanger />", () => {
  test("should render LanguageChanger component", () => {
    const wrapper = render(<LanguageChanger />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
