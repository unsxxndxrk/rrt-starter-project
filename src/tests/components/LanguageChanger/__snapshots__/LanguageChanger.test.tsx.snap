// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageChanger /> should render LanguageChanger component 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root"
  >
    <div
      class="makeStyles-langWrap"
    >
      <div
        class="makeStyles-lang active"
      >
        en
      </div>
      |
    </div>
    <div
      class="makeStyles-langWrap"
    >
      <div
        class="makeStyles-lang"
      >
        ru
      </div>
    </div>
  </div>
</DocumentFragment>
`;
